mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -935,8 +909,10 @@ struct ConversionPatternRewriterImpl : public RewriterBase::Listener {
   /// to modify/access them is invalid rewriter API usage.
   SetVector<Operation *> replacedOps;
 
-  /// A set of all unresolved materializations.
-  DenseSet<Operation *> unresolvedMaterializations;
+  /// A mapping of all unresolved materializations (UnrealizedConversionCastOp)
+  /// to the corresponding rewrite objects.
+  DenseMap<Operation *, UnresolvedMaterializationRewrite *>
```

Comment:
Can the key be directly `UnrealizedConversionCastOp` ?

Reply:
This set may contain ops that were already erased. (When ops from the set are accessed, there is an extra check to see if the op was erased.) I wanted to avoid calling `op.getOperation()` on an erased `UnrealizedConversionCastOp` (it would probably work, but looks dangerous). (The function that checks if an op was erased expects an `Operation *`.)

Reply:
`UnrealizedConversionCastOp` is implicit convertible to `Operation*`, and it is never the `UnrealizedConversionCastOp` class that is deleted, it is the `Operation *`. Think of `UnrealizedConversionCastOp` as another name for `Operation *` :)

So if it is safe to store a `Operation *` it should be equally safe to store an `UnrealizedConversionCastOp`.